src/System.IO.FileSystem/src/System/IO/FileSystem.DirectoryCreation.Windows.cs

```diff
@@ -0,0 +1,129 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Win32.SafeHandles;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.IO;
+using System.Text;
+
+#if MS_IO_REDIST
+namespace Microsoft.IO
+#else
+namespace System.IO
+#endif
+{
+    internal static partial class FileSystem
+    {
+        public static unsafe void CreateDirectory(string fullPath, byte[] securityDescriptor = null)
```

Comment:
@JeremyKuhne the `string fullPath` is where I wanted to ask you about `string.IsNullOrWhitespace`, not the `DirectorySecurity` object. We don't check it. Should we?

Reply:
DirectoryInfo does.